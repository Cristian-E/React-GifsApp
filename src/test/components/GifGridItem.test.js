import { shallow } from "enzyme"
import { GifGridItem } from "../../Components/GifGridItem"
import React from 'react'


describe('Pruebas en <GifGridItem />', () => {

    const url = "https://localhost/img.jsp"
    const title = "Soy un titulo"
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('debe renderizar el componente', () => {


        expect(wrapper).toMatchSnapshot()
    })

    test('debe mostrar el title', () => {

        const p = wrapper.find("p")
        expect(p.text().trim()).toBe(title)

    })

    test('debe de tener el url y alt de los props', () => {

        const img = wrapper.find("img")

        expect(img.prop("src")).toBe(url)
        expect(img.prop("alt")).toBe(title)
    })

    test('debe de tener animate__bounce', () => {
        const div = wrapper.find("div")

        expect(div.prop("className").includes("animate__bounce")).toBe(true)


    })




})