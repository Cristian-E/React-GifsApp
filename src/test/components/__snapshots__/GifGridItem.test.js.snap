// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe renderizar el componente 1`] = `
<div
  className="card animate__animated animate__bounce "
>
  <img
    alt="Soy un titulo"
    src="https://localhost/img.jsp"
  />
  <p>
    Soy un titulo
  </p>
</div>
`;
